src/actions/move.py

```diff
@@ -1,13 +1,35 @@
-from .action import Action
 from src.common.enums import Direction
-from src.objects.agent import Agent
+from src.common.point import Point
+from .action import Action
 
 
 class Move(Action):
-    def __init__(self, agent: Agent, direction: Direction) -> None:
-        super().__init__(agent.properties.id, agent.properties.id)
+    '''
+    Simple action for changing position of `Agent`. It does not log anything
+    in case the movement was not possible(destination is not walkable etc).
+    '''
+
+    def __init__(self, agent, direction: Direction) -> None:
+        super().__init__()
+        self.properties.agent_id = agent.properties.id
+        self.agent = agent
         self.properties.direction: Direction = direction
 
-    def execute(self) -> str:
-        self.agent.properties.position += self.properties.direction.value
-        return super().execute()
+    def execute(self, scene) -> str:
+        new_position: Point = self.agent.properties.position + \
+            self.properties.direction.value
+
+        # Check if the destination is a `field` in the scene, and if it is -
+        # - it must be walkable
+        if scene[new_position] is not None and scene[new_position][0].walkable:
```

Comment:
I think `scene` should have a method for checking if destination is plausible.
Somethin like `scene.is_field(new_pos)`.

Reply:
Sure, as I said - it is a subject to change, but I think that a `property` is a better way of describing an `Object` rather than creating come pointless function doing just `return object.walkable`.